PR: https://github.com/odoo/odoo/pull/

From: 508e9dd9fe79379540fb90049a70780f89c5ad43
From: Olivier Dony
Date: 2017-03-28 12:45:29

Structural Changes: 4
Total Changes: 17

[IMP] hr*: show correct fields to employees

================================= pseudo patch: =================================

--- a/addons/hr/models/hr.py
+++ b/addons/hr/models/hr.py
@@ -111,26 +111,27 @@ class Employee(models.Model):
     active = fields.Boolean('Active', related='resource_id.active', default=True, store=True)
 
     country_id = fields.Many2one('res.country', string='Nationality (Country)')
-    birthday = fields.Date('Date of Birth')
-    ssnid = fields.Char('SSN No', help='Social Security Number')
-    sinid = fields.Char('SIN No', help='Social Insurance Number')
-    identification_id = fields.Char(string='Identification No')
+    birthday = fields.Date('Date of Birth', groups="hr.group_hr_user")
+    ssnid = fields.Char('SSN No', help='Social Security Number', groups="hr.group_hr_user")
+    sinid = fields.Char('SIN No', help='Social Insurance Number', groups="hr.group_hr_user")
+    identification_id = fields.Char(string='Identification No', groups="hr.group_hr_user")
     gender = fields.Selection([
         ('male', 'Male'),
         ('female', 'Female'),
         ('other', 'Other')
-    ])
+    ], groups="hr.group_hr_user")
     marital = fields.Selection([
         ('single', 'Single'),
         ('married', 'Married'),
         ('widower', 'Widower'),
         ('divorced', 'Divorced')
-    ], string='Marital Status')
+    ], string='Marital Status', groups="hr.group_hr_user")
     department_id = fields.Many2one('hr.department', string='Department')
     address_id = fields.Many2one('res.partner', string='Work Address')
     address_home_id = fields.Many2one('res.partner', string='Home Address')
     bank_account_id = fields.Many2one('res.partner.bank', string='Bank Account Number',
-        domain="[('partner_id', '=', address_home_id)]", help='Employee bank salary account')
+        domain="[('partner_id', '=', address_home_id)]", help='Employee bank salary account',
+        groups="hr.group_hr_user")
     work_phone = fields.Char('Work Phone')
     mobile_phone = fields.Char('Work Mobile')
     work_email = fields.Char('Work Email')
@@ -141,7 +142,7 @@ class Employee(models.Model):
     child_ids = fields.One2many('hr.employee', 'parent_id', string='Subordinates')
     coach_id = fields.Many2one('hr.employee', string='Coach')
     job_id = fields.Many2one('hr.job', string='Job Title')
-    passport_id = fields.Char('Passport No')
+    passport_id = fields.Char('Passport No', groups="hr.group_hr_user")
     color = fields.Integer('Color Index', default=0)
     city = fields.Char(related='address_id.city')
 
